frontend/src/components/Form.tsx

```diff
@@ -0,0 +1,95 @@
+import { useState } from 'react'
+import { FormField } from './FormField'
+
+export interface FormData {
+  name: string
+  age: string
+  favoriteColor: string
+  preferredContactMethod: string
+}
+
+export const Form = () => {
+  const [formData, setFormData] = useState({
+    name: '',
+    age: '',
+    favoriteColor: '',
+    preferredContactMethod: 'Email',
+  })
+  const contactMethods = ['Email', 'Phone']
+
+  const handleSubmitForm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    if (formData.name === '' || formData.age === '' || formData.favoriteColor === '') {
+      return
+    }
+
+    try {
+      const response = await fetch('http://gateway:8080/api/submit', {
```

Comment:
Ok lets maybe define extarnal file with api client or generate it from swagger

Reply:
Anyway wiil be nice to extract http://gateway:8080 into some global variable, the url is wrong localhost/api/submit is proper using docker compose ( ;) )